corrections/BaseCorrector.py

```diff
@@ -394,23 +424,41 @@ def save_lightcurve(self, lc, output_folder=None):
 		}.get(self.__class__.__name__)
 
 		# Decide where to save the finished lightcurve:
-		if output_folder is None:
+		if self.output_folder is None:
 			output_folder = self.input_folder
+		else:
+			output_folder = self.output_folder
 
 		# Get the filename of the original file from the task:
 		fname = lc.meta.get('task').get('lightcurve')
 
 		if fname.endswith('.fits') or fname.endswith('.fits.gz'):
-			#if output_folder != self.input_folder:
-			save_file = os.path.join(output_folder, os.path.dirname(fname), 'corr-' + os.path.basename(fname))
+			
+			if CorrMethod == 'CBV':
+				filename = os.path.basename(fname).replace('-tasoc_lc', '-tasoc-cbv_lc')
+			if CorrMethod == 'Ensemble':
+				filename = os.path.basename(fname).replace('-tasoc_lc', '-tasoc-ens_lc')
+			if CorrMethod == 'KASOC Filter':
+				filename = os.path.basename(fname).replace('-tasoc_lc', '-tasoc-kf_lc')
+			
+			
+			
+			if output_folder != self.input_folder:
+				save_file = os.path.join(output_folder, filename)
+			else:
+				save_file = os.path.join(output_folder, os.path.dirname(fname), filename)
+							
 			shutil.copy(os.path.join(self.input_folder, fname), save_file)
+			
+			# Change permission of copied file to allow the addition of the corrected lightcurve
+			os.chmod(save_file, 0o640) 
 
 			# Open the FITS file to overwrite the corrected flux columns:
-			with fits.open(save_file, mode='update') as hdu:
+			with fits.open(os.path.join(self.input_folder,fname), mode='update') as hdu:
 				# Overwrite the corrected flux columns:
 				hdu['LIGHTCURVE'].data['FLUX_CORR'] = lc.flux
-				hdu['LIGHTCURVE'].data['FLUX_CORR_ERR'] = lc.flux_err
-				hdu['LIGHTCURVE'].data['QUALITY'] = lc.quality
+				#hdu['LIGHTCURVE'].data['FLUX_CORR_ERR'] = lc.flux_err
```

Comment:
Are these supposed to remain commented out in the final version?

Reply:
No... Again, an example of a change in this PR that we should revert